### PR TITLE
Remove extra convolutions in TCN

### LINKED ISSUE #470: [BUG] Extra convolutions in TCN
**Describe the bug**
There are extra 1x1 convolutions in the first and the last residual blocks in a case of the fixed number of channels, say input_dim=4, num_filters=4, output_dim=4. There are no need to add such convolutions because the residual summation can be directly performed to the output of the second convolution.

**To Reproduce**
```python
from numpy.random import randn
from darts.models import TCNModel
train_sample = (randn(10, 4), None, randn(10, 4))
model = TCNModel(input_chunk_length=10, output_chunk_length=9, num_layers=2, num_filters=4)
model._create_model(train_sample)
```

**Expected behavior**
There are no extra convolutions

**System**
 - darts version 0.11.0

### PR DESCRIPTION
**Pull request by vpozdnyakov**
Fixes #470. Add the 1x1 convolution in a case of different number of channels between input and output of a residual block

### CODE REVIEW

File: darts/models/tcn_model.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def __init__(self,
         if weight_norm:
             self.conv1, self.conv2 = nn.utils.weight_norm(self.conv1), nn.utils.weight_norm(self.conv2)
 
-        if nr_blocks_below == 0 or nr_blocks_below == num_layers - 1:
+        if nr_blocks_below in {0, num_layers - 1} and input_dim != output_dim:
```
Comment (pennfranc) [inline darts/models/tcn_model.py:87]: Actually do you think we could simplify this to
```suggestion
        if  input_dim != output_dim:
```

File: darts/models/tcn_model.py [hunk 2]
```diff
@@ -103,7 +103,8 @@ def forward(self, x):
         x = self.dropout_fn(x)
 
         # add residual
-        if self.nr_blocks_below in {0, self.num_layers - 1}:
+        if (self.nr_blocks_below in {0, self.num_layers - 1} and
```
Comment (pennfranc) [inline darts/models/tcn_model.py:106]: And also here just check, `self.conv1.in_channels != self.conv2.out_channels`?

File: darts/models/tcn_model.py (hunk 1 above)
Comment (vpozdnyakov) [reply darts/models/tcn_model.py:87]: Agree! Looks super simple and clear

**Review by hrzn — APPROVED** [review]
Thanks!

**Review by pennfranc — APPROVED** [review]
Thanks for polishing our TCN!

**Pull request merged**